core/src/test/java/org/testcontainers/containers/wait/internal/InternalCommandPortListeningCheckTest.java

```diff
@@ -1,25 +1,53 @@
 package org.testcontainers.containers.wait.internal;
 
 import com.google.common.collect.ImmutableSet;
-import org.junit.Rule;
+import org.junit.After;
+import org.junit.Before;
 import org.junit.Test;
-import org.testcontainers.containers.BindMode;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
 import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.images.builder.ImageFromDockerfile;
 
+import static java.util.Arrays.asList;
 import static org.rnorth.visibleassertions.VisibleAssertions.assertFalse;
 import static org.rnorth.visibleassertions.VisibleAssertions.assertTrue;
 
+@RunWith(Parameterized.class)
 public class InternalCommandPortListeningCheckTest {
 
-    // Linking a custom configuration into the container so that nginx is listening on port 8080. This is necessary to prove
-    // that the command formatting uses the correct casing for hexadecimal numbers (i.e. 1F90 and not 1f90).
-    @Rule
-    public GenericContainer nginx = new GenericContainer<>("nginx:1.9.4")
-            .withClasspathResourceMapping("nginx_on_8080.conf", "/etc/nginx/conf.d/default.conf", BindMode.READ_ONLY);
+    @Parameterized.Parameter
+    public String dockerfile;
+
+    @Parameterized.Parameters(name = "{index} - {0}")
+    public static Iterable<Object[]> data() {
+        return asList(
+            new Object[][]{
+                {"internal-port-check-dockerfile/Dockerfile-tcp"},
+                {"internal-port-check-dockerfile/Dockerfile-nc"},
+                {"internal-port-check-dockerfile/Dockerfile-bash"},
+            });
+    }
+
+    public GenericContainer container;
+
+    @Before
+    public void setUp() {
+        container = new GenericContainer(new ImageFromDockerfile()
```

Comment:
You can keep `@Rule` if you move the container creation to constructor (JUnit supports parameterized constructors if I remember correctly)

Reply:
Tried this already and it results in `NullPointerException` due to `dockerfile` being null

Reply:
it is of course null because it's a field. You need to delete the field and replace it with constructor's parameter

Reply:
Hmm ok after some trial and error I figured out how to use a @Rule here. Hope it's actually what you referred to.